File: ht_web/lib/business/course/model/course.dart
final class Course {
  final String id, name, description, mode, dueDate;
  final int duration;
  final double price;
  const Course({
    required this.id,
    required this.name,
    this.description = 'NA',
    required this.mode,
    required this.dueDate,
    required this.duration,
    required this.price,
  });

  factory Course.fromJSON({required Map<String, dynamic> response}) {
    return Course(
      id: response['_id'],
      name: response['name'],
      description: response['description'],
      mode: response['mode'],
      dueDate: response['due_date'],
      duration: response['duration'],
      price: double.parse(response['price']),
    );
  }
}

File: backend/model/course.js
function Course(sequelize, DataTypes){
    return sequelize.define('COURSE', {
        _id: { type: DataTypes.STRING(37), allowNull: false, primaryKey: true },
        name: { type: DataTypes.STRING(50), allowNull: false },
        description: { type: DataTypes.TEXT, allowNull: true },
        duration: { type: DataTypes.INTEGER, allowNull: false },
        price: { type: DataTypes.DECIMAL(10,2), allowNull: false },
        mode: { type: DataTypes.ENUM('RECORDED', 'LIVE'), allowNull: false },
        due_date: { type: DataTypes.DATE }
    },{ timestamps: false, freezeTableName: true });
}

export default Course;